### ISSUE TITLE
Can I add softmax layer at the top?

### ISSUE THREAD

**Issue by didw** [jcjohnson/torch-rnn#19]
Hi jcjohnson,
Thanks for sharing the code. It's really helpful.
I want to add `nn.LogSoftMax()` at the top layer but couldn't find way to do it because batch and depth size.
Isn't that possible to add `nn.LogSoftMax` module?

Thanks in advance.

**Comment by jcjohnson** [comment]
It should be easy to do, you'll just need to add a View before and after the LogSoftMax, like this:

(N, T, D) -> View -> (N*T, D) -> LogSoftMax -> View -> (N, T, D)

This is the trick I use to stick BatchNormalization between layers:

https://github.com/jcjohnson/torch-rnn/blob/master/LanguageModel.lua#L53

The only gotcha is that you need to tell the View what N and T are on every forward pass, like this:

https://github.com/jcjohnson/torch-rnn/blob/master/LanguageModel.lua#L85

This is a common pattern, so it might make sense to write a generic TemporalAdapter to handle this. It would be easy to do.

**Comment by jcjohnson** [comment]
I went ahead and implemented a generic TemporalAdapter that you can easily use to solve this problem; this had the side effect of cleaning up the LanguageModel quite a bit. Now to add a LogSoftMax you'd just need to add nn.TemporalAdapter(nn.LogSoftMax()) to the network, which is now how BatchNormalization and the final affine layer work:

https://github.com/jcjohnson/torch-rnn/blob/master/LanguageModel.lua#L49

**Comment by didw** [comment]
Thanks jcjohnson,
Thanks for quick response.
I'll test it right away.
